src/aiortc/rtcdtlstransport.py

```diff
@@ -431,17 +431,15 @@ async def start(self, remoteParameters: RTCDtlsParameters) -> None:
 
         # check remote fingerprint
         x509 = self._ssl.get_peer_certificate()
-        remote_fingerprint = certificate_digest(x509)
-        fingerprint_is_valid = False
         for f in remoteParameters.fingerprints:
-            if (
-                f.algorithm.lower() == "sha-256"
-                and f.value.lower() == remote_fingerprint.lower()
-            ):
-                fingerprint_is_valid = True
+            try:
+                fingerprint = x509.digest(f.algorithm).decode("ascii").upper()
```

Comment:
probably good to check the algorithm against an allow list containing sha-256, sha-384 and sha-512 first.
Otherwise this would bring back sha-1 which nobody wants in 2024 ;-)

Reply:
I'm not sure if `SHA-1` is an issue in this context (it's just the certificate fingerprint) but a configurable whitelist would not hurt. But that would be a separate pull request and a bigger change?